cmd/alertmanager/main.go

```diff
@@ -478,6 +478,8 @@ func run() int {
 			timeIntervals[ti.Name] = ti.TimeIntervals
 		}
 
+		intervener := timeinterval.NewIntervener(timeIntervals)
```

Comment:
I started moving the loop above into `timeinterval` as well but quickly realised that we have an import problem - this is part of a larger problem and I'll address it in a separate PR.

I don't understand why the marshalling of config -> struct happens with structs that live in `timeinterval` when this is generally done as part of the`config` package.